h5netcdf/core.py

```diff
@@ -462,6 +514,32 @@ def __repr__(self):
                   % (self._cls_name, self.name, len(self)))
         return '\n'.join([header] + self._repr_body())
 
+    def resize_dimension(self, dimension, size):
+        """
+        Resize a dimension to a certain size.
+
+        This will pad with zeros where necessary.
+        """
+        if self.dimensions[dimension] is not None:
+            raise ValueError("Only unlimited dimensions can be resized.")
+
+        # Resize the dimension.
+        self._current_dim_sizes[dimension] = size
+        self._dim_order[dimension] = len(self._dim_order)
```

Comment:
Why does `dim_order` get changed when you resize?

Reply:
Yea this is a mistake.